### PR TITLE
entgql: export templates files and sync with internal version

### PR DESCRIPTION
**Pull request by a8m**

### CODE REVIEW

File: entgql/template.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	AllTemplates = []*template.Template{
+		CollectionTemplate,
+		EnumTemplate,
+		NodeTemplate,
+		PaginationTemplate,
+		TransactionTemplate,
+	}
+)
+
+//go:generate go run github.com/go-bindata/go-bindata/go-bindata -o=internal/bindata.go -pkg=internal -modtime=1 ./template
```
Comment (a8m) [inline entgql/template.go:41]: This will be replaced with `//go:embed` in the future releases.

**Review by alexsn — APPROVED** [review]
LGTM

File: entgql/template.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	"github.com/facebook/ent/entc/gen"
+	_ "github.com/go-bindata/go-bindata"
+)
+
+var (
+	// CollectionTemplate adds fields collection support using auto eager-load ent edges.
+	// More info can be found here: https://spec.graphql.org/June2018/#sec-Field-Collection.
+	CollectionTemplate = parse("template/collection.tmpl")
+
+	// EnumTemplate adds a template for implementing the MarshalGQL methods for enums.
```
Comment (alexsn) [inline entgql/template.go:17]:
```suggestion
	// EnumTemplate adds a template implementing MarshalGQL/UnmarshalGQL methods for enums.
```

File: entgql/template.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	EnumTemplate = parse("template/enum.tmpl")
+
+	// NodeTemplate implements the GraphQL Node interface for all types.
```
Comment (alexsn) [inline entgql/template.go:20]:
```suggestion
	// NodeTemplate implements the Relay Node interface for all types.
```

File: entgql/template.go [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+	NodeTemplate = parse("template/node.tmpl")
+
+	// PaginationTemplate adds pagination support according to the GraphQL Cursor Connections Spec.
```
Comment (alexsn) [inline entgql/template.go:23]:
```suggestion
	// PaginationTemplate adds pagination support according to the Relay Cursor Connections Spec.
```

**Pull request merged**